### PR TITLE
Make readystatechange fire more often (fixes #13481)

### LINKED ISSUE #13481: Dispatch XHR readystatechange events more often
https://github.com/whatwg/xhr/commit/8efe6aad48b2824dbd7cb79836946f94d674ad4d

### PR DESCRIPTION
**Pull request by mathieuh**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13481 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13485)

### CODE REVIEW

**Comment by jdm** [comment]
The failing test is the one mentioned in https://github.com/w3c/web-platform-tests/issues/3834.

**Comment by mathieuh** [comment]
Hi @jdm,

What exactly do you want to do with this test, can't we just disable it?

**Comment by jdm** [comment]
As part of modifying our implementation, we should update the test to reflect the new specification. We should presumably pass it after modifying it, too.

**Comment by mathieuh** [comment]
I managed to update the test thanks to @nox

**Comment by jdm** [comment]
We should change the title of the test as well, so as not to confuse matters.

**Comment by mathieuh** [comment]
@jdm Sure, my bad, it's done.

File: components/script/dom/xmlhttprequest.rs [hunk 1]
```diff
@@ -974,9 +974,15 @@ impl XMLHttpRequest {
...
                     if self.ready_state.get() == XMLHttpRequestState::HeadersReceived {
-                        self.change_ready_state(XMLHttpRequestState::Loading);
-                        return_if_fetch_was_terminated!();
+                        self.ready_state.set(XMLHttpRequestState::Loading);
                     }
+                    let global = self.global();
+                    let event = Event::new(global.r(),
+                           atom!("readystatechange"),
+                           EventBubbles::DoesNotBubble,
+                           EventCancelable::Cancelable);
```
Comment (jdm) [inline components/script/dom/xmlhttprequest.rs]: nit: please indent these lines to line up with `global.r()`.

**Review by jdm — CHANGES_REQUESTED** [review]
With the one indentation change, this looks fine to merge.

File: components/script/dom/xmlhttprequest.rs (hunk 1 above)
Comment (mathieuh) [inline components/script/dom/xmlhttprequest.rs]: Updated, tell me if it's ok like that.

Comment (jdm) [inline components/script/dom/xmlhttprequest.rs]: I actually meant indenting 981, 982, and 983 to match the position of `global.r()` on line 980 ;)

Comment (nox) [inline components/script/dom/xmlhttprequest.rs]: We also use that style of indentation in other places, and I think we should prefer it given it reduces the rightwards drift with longer expressions. Are you still blocking on this or are you ok with the current code @jdm?

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by jdm** [comment]
@bors-servo: retry
* #13563

**Pull request merged**